### PR TITLE
A simple Cross-Unity Project Dependency Relationship

### PR DESCRIPTION
**Pull request by andreiborodin**
This change sets up support for a Unity-to-Unity project dependency relationship and adds a sample shwocasing this support.

### CODE REVIEW

File: Samples/IntegratedDependencies.Unity/Assets/GeneralComponent/GeneralComponentType.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿using CommonLibrary;
+using UnityEngine;
```
Comment (chrisfromwork) [inline Samples/IntegratedDependencies.Unity/Assets/GeneralComponent/GeneralComponentType.cs]: add license header

File: Samples/IntegratedDependencies.Unity/Assets/GeneralComponent/GeneralComponentType.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    private CommonComponent commonComponent = new CommonComponent();
+    private void Start()
+    {
+        Debug.Log(commonComponent.GetData());
+    }
+
+    public string GetThisDataToo()
+    {
+        return "IntegartedDependencies Project + {commonComponent.GetData()}";
+    }
```
Comment (chrisfromwork) [inline Samples/IntegratedDependencies.Unity/Assets/GeneralComponent/GeneralComponentType.cs]: nit: this should be $""

File: Samples/CrossUnityDependencies.Unity/Assets/Dependencies.msb4u.csproj [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    Examples of how you can modify this file:
+    - Add NuGet package references:
+        <ItemGroup>
+          <PackageReference Include="Newtonsoft.Json" Version="11.0.1" />
+        </ItemGroup>
+    - Add external C# project references:
+      <ItemGroup>
+        <ProjectReference Include="..\..\..\ExternalLib\ExternalLib.csproj" />
+      </ItemGroup>
+  -->
```
Comment (chrisfromwork) [inline Samples/CrossUnityDependencies.Unity/Assets/Dependencies.msb4u.csproj]: do you think we benefit from having this much content in all generated files? It kind of seems like if I was only shown a reference example this file would be less intimidating for someone new to csproj definitions.

Comment (chrisfromwork) [reply Samples/CrossUnityDependencies.Unity/Assets/Dependencies.msb4u.csproj]: reducing the amount of content in this file will make it easier to read in things like notepad

---
In reply to: [367709527](https://github.com/microsoft/MSBuildForUnity/pull/108#discussion_r367709527) [](ancestors = 367709527)

Comment (andreiborodin) [reply Samples/CrossUnityDependencies.Unity/Assets/Dependencies.msb4u.csproj]: I will keep it for now, but will consider it in the future.

---
In reply to: [367709703](https://github.com/microsoft/MSBuildForUnity/pull/108#discussion_r367709703) [](ancestors = 367709703,367709527)

**Pull request merged**